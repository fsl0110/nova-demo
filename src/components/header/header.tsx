import React, { FC, HTMLAttributes } from "react";
import classNames from "classnames";
import { NavLink } from "react-router-dom";

export interface Props extends HTMLAttributes<HTMLDivElement> {
  /** This component does not support custom children. */
  children?: never;
  /** Array of navigation links shown in header */
  navLinks: Array<{ name: string; path: string }>;
  /** Align of the navLinks within the header */
  navLinksAlign?: "center" | "right";
}

export const Header: FC<Props> = ({
  navLinks,
  navLinksAlign,
  className,
  ...rest
}) => {
  const rootClasses = classNames("header", {
    [`nav--${navLinksAlign}`]: !!navLinksAlign
  });

  return (
    <header className={rootClasses} {...rest}>
      <nav>
        {navLinks.map((link: { name: string; path: string }) => (
          <NavLink
            className="link"
            key={link.path}
            to={`/${link.path}`}
            activeClassName="link--current"
          >
            {link.name}
          </NavLink>
        ))}
      </nav>
    </header>
  );
};

export default Header;
