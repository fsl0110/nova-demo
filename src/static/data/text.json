{
  "reports": {
    "topic": "Food recall enforcement reports since 2012",
    "teaser": "",
    "text": "This is the openFDA API endpoint for all food product recalls monitored by the FDA. When an FDA-regulated product is either defective or potentially harmful, recalling that product—removing it from the market or correcting the problem—is the most effective means for protecting the public.",
    "ChartTitle": "Enforcement reports over time"
  },
  "initiators": {
    "topic": "The vast majority of food recalls are voluntary",
    "teaser": "",
    "text": "Manufacturers and/or distributors may initiate a recall at any time to fulfill their responsibility to protect the public health from products that present a risk of injury or gross deception, or are otherwise defective. Firms may also initiate a recall following notification of a problem by FDA or a state agency, in response to a formal request by FDA, or as ordered by FDA.",
    "chartTitle": "Enforcement initiators"
  }
}
