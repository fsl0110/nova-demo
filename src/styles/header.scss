.header {
  width: 100%;
  display: flex;
  align-items: top;

  padding-bottom: 0.8rem;

  margin-top: -0.8rem;
}

.nav {
}

.nav--center {
  justify-content: center;
}

.nav--right {
  justify-content: flex-end;
}

.link {
  text-decoration: none;
  color: #95ceff;
  font-size: 1.2rem;
}

.link--current {
  color: rgb(32, 32, 32);
  font-weight: bold;
}

.link + .link {
  margin-left: 1rem;
}
