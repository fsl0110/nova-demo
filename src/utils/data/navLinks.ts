export const navLinks: Array<{ name: string; path: string }> = [
  {
    name: "Highcharts",
    path: "highcharts"
  },
  {
    name: "D3.js",
    path: "d3-js"
  }
];
