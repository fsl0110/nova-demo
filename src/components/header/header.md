### Header Component

#### NavLinks default align left

```jsx
const MemoryRouter = require("react-router").MemoryRouter;
import { navLinks } from "../../utils/data/navLinks";
console.log("test:", navLinks);
<MemoryRouter>
  <Header navLinks={navLinks} />
</MemoryRouter>;
```

#### NavLinks default align center

```jsx
const MemoryRouter = require("react-router").MemoryRouter;
import { navLinks } from "../../utils/data/navLinks";

<MemoryRouter>
  <Header navLinks={navLinks} navLinksAlign="center" />
</MemoryRouter>;
```

#### NavLinks default align right

```jsx
const MemoryRouter = require("react-router").MemoryRouter;
import { navLinks } from "../../utils/data/navLinks";

<MemoryRouter>
  <Header navLinks={navLinks} navLinksAlign="right" />
</MemoryRouter>;
```
