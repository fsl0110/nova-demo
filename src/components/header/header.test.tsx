import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Header, Props } from "./header";
import { findByTestAttr } from "../../utils/tests";
import { navLinks } from "../../utils/data/navLinks";

const testCases: Array<[string, Props]> = [
  ["navLinks default left", { navLinks }],
  ["navLinks center", { navLinks, navLinksAlign: "center" }],
  ["navLinks right", { navLinks, navLinksAlign: "right" }]
];

describe("<Header>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(toJson(shallow(<Header {...options} />))).toMatchSnapshot();
    });
  });
});
