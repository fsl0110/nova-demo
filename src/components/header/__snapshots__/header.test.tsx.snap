// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> navLinks center 1`] = `
<header
  className="header nav--center"
>
  <nav>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="highcharts"
      to="/highcharts"
    >
      Highcharts
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="d3-js"
      to="/d3-js"
    >
      D3.js
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="linebarchart"
      to="/linebarchart"
    >
      LinebarChart
    </NavLink>
  </nav>
</header>
`;

exports[`<Header> navLinks default left 1`] = `
<header
  className="header"
>
  <nav>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="highcharts"
      to="/highcharts"
    >
      Highcharts
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="d3-js"
      to="/d3-js"
    >
      D3.js
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="linebarchart"
      to="/linebarchart"
    >
      LinebarChart
    </NavLink>
  </nav>
</header>
`;

exports[`<Header> navLinks right 1`] = `
<header
  className="header nav--right"
>
  <nav>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="highcharts"
      to="/highcharts"
    >
      Highcharts
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="d3-js"
      to="/d3-js"
    >
      D3.js
    </NavLink>
    <NavLink
      activeClassName="link--current"
      className="link"
      key="linebarchart"
      to="/linebarchart"
    >
      LinebarChart
    </NavLink>
  </nav>
</header>
`;
